static/js/is-staging.js

```diff
@@ -1,9 +1,19 @@
+import escapeStringRegexp from 'escape-string-regexp';
+
 /**
  * Determines whether the current url is staging or production.
  */
 export function isStaging(stagingDomains) {
   const defaultStagingDomains = ['127.0.0.1', 'localhost', 'office.yext.com'];
-  const _stagingDomains = defaultStagingDomains.concat(stagingDomains);
-  const currentUrl = window.location.href;
-  return _stagingDomains.some(domain => currentUrl.includes(domain));
-}
+  const _stagingDomains = stagingDomains
+    ? defaultStagingDomains.concat(stagingDomains)
+    : defaultStagingDomains;
+  const currentDomain = window.location.hostname;
+
+  return _stagingDomains.some(domain => {
+    const regexEscapedDomain = escapeStringRegexp(domain);
+    const isSubdomainRegex = new RegExp(`[A-Za-z0-9]?[A-Za-z0-9\-]{0,61}[A-Za-z0-9]?\.${regexEscapedDomain}`);
```

Comment:
Why 0 to 61 instances?

Reply:
The RFC for subdomains says they can contain a maximum of 63 characters. The leading and trailing characters must be alpha numeric.